README.md

```diff
@@ -22,72 +22,301 @@ terraform {
+      public_subnets = ["10.2.0.0/24"]
+    },
+    {
+      region         = "ap-northeast-2"
+      cidr           = "10.3.0.0/16"
+      public_subnets = ["10.3.0.0/24"]
     }
   ]
+
   firewall_rules = [
     {
+      name    = "postgres"
       port    = 5432
       sources = ["0.0.0.0/0"]
     },
-    {
-      port    = 22
-      sources = ["0.0.0.0/0"]
-    }
   ]
+
+  depends_on = [pgedge_cloud_account.example]
+}
+
+### Updating a Cluster
+
+To update an existing cluster, such as adding or removing nodes, follow these steps:
+
+- **Add or remove nodes**: Modify the `nodes` block.
+- **Update regions and networks**: If you're adding or removing nodes, you must also update the corresponding `regions` and `networks` blocks for those nodes.
+- **Manage backup stores**: Adjust the `backup_store_ids` array if necessary.
+
+For example, when removing a node, ensure the corresponding region and network are also removed.
+
+---
+
+**Example: Removing a Node from the Cluster**
+
+To remove a node, you must also update the corresponding `regions` and `networks`:
+
+```hcl
+nodes = [
+  {
+    name          = "n1"
+    region        = "ap-northeast-1"
+    instance_type = "t4g.medium"
+    volume_size   = 20
+    volume_type   = "gp2"
+  },
+  {
+    name          = "n3"
+    region        = "ap-northeast-2"
+    instance_type = "t4g.medium"
+    volume_size   = 20
+    volume_type   = "gp2"
+  }
+]
+
+regions = ["ap-northeast-1", "ap-northeast-2"]
+
+networks = [
+  {
+    region         = "ap-northeast-1"
+    cidr           = "10.1.0.0/16"
+    public_subnets = ["10.1.0.0/24"]
+  },
+  {
+    region         = "ap-northeast-2"
+    cidr           = "10.3.0.0/16"
+    public_subnets = ["10.3.0.0/24"]
+  }
+]
+```
+
+This ensures that all associated regions, nodes, and networks are synchronized when making changes.
+
+---
+
+### Database Resource with Update Support
+
+```hcl
+resource "pgedge_database" "example_db" {
+  name       = "exampledb"
+  cluster_id = pgedge_cluster.example.id
+
+  options = [
+    "install:northwind",
+    "rest:enabled",
+    "autoddl:enabled",
+    "cloudwatch_metrics:enabled"
+  ]
+
+  extensions = {
+    auto_manage = true
+    requested = [
+      "postgis",
+      "vector"
+    ]
+  }
+
   nodes = [
     {
-      name              = "n1"
-      region            = "us-west-1"
-      availability_zone = "us-west-1a"
-      instance_type     = "t4g.medium"
-      volume_size       = 20
-      volume_type       = "gp2"
+      name = "n1"
     },
     {
-      name              = "n2"
-      region            = "us-west-2"
-      availability_zone = "us-west-2a"
-      instance_type     = "t4g.medium"
-      volume_size       = 20
-      volume_type       = "gp2"
+      name = "n2"
+    },
+    {
+      name = "n3"
     }
   ]
-}
 
-# Create a database
-resource "pgedge_database" "example" {
-  name       = "exampledb"
-  cluster_id = pgedge_cluster.main.id
+  backups = {
+    provider = "pgdump"
```

Comment:
Let's show pgbackrest as the default example here for Enterprise, with the "standard" default config we provision in the UI (one config, daily full, hourly incremental)